### DISCUSSION TITLE
Passagem de Args com modular.

### DISCUSSION THREAD

**Discussion by rsynQ** [Flutterando/forum#68] [Perguntas e Respostas]
Olá, estou construindo uma aplicação que consume uma API de terceiro com DIO e Mobx.

Estou com dificuldade de definir melhor fluxo para sempre ter esses dados atualizados, na HOME meu controller está fazendo chamada a api e já deixando os dados atualizados. Agora preciso que ao clicar em um item da lista para ver os detalhes essas informações também fique sempre atualizadas. Qual melhor forma de fazer isso? pois atualmente consigo visualizar os detalhes mais as informações não são atualizadas com frequência.

HomePage
```
 InkWell(
            onTap: (){
              _controller.goToPageMatchDetail(_match);
            },


GameModule
```
```
class GameModule extends Module {

   @override
   final List<Bind> binds = [];

   @override
   final List<ModularRoute> routes = [
    ChildRoute(Modular.initialRoute, child: (context, args) => GamePage(match: args.data))
   ];

}

```

**Comment by eduardoflorence** [comment]
@rsynQ não deu para entender perfeitamente onde está seu problema, mas vou tentar ajudar.
Como está usando MobX, além da sua lista ser um Store, cada item desta lista também deve ser um store para ficar observável também ao fazer uma alteração. 

```dart
import 'package:mobx/mobx.dart';

part 'todo.g.dart';

class Todo = _Todo with _$Todo;

abstract class _Todo with Store {
  _Todo(this.description);

  @observable
  String description = '';

  @observable
  bool done = false;
}
```

```dart
import 'package:mobx/mobx.dart';

part 'todo_list.g.dart';

class TodoList = _TodoList with _$TodoList;

abstract class _TodoList with Store {
  @observable
  ObservableList<Todo> todos = ObservableList<Todo>();
}
```

Por fim, no módulo, injeta a store de lista via construtor na página de detalhes e passa como argumento na navegação apenas o índice do item na lista.
